### PR TITLE
Fix issue 646: Rule L036 misbehaves when there's trailing whitespace after "SELECT"

### PR DESCRIPTION
**Pull request by barrywhart**
Addresses #646

### CODE REVIEW

File: src/sqlfluff/core/rules/std.py [hunk 1]
```diff
@@ -3462,11 +3462,14 @@ def _get_indexes(segment):
                 cnt_select_targets += 1
                 if first_select_target_idx == -1:
                     first_select_target_idx = fname_idx
-            if seg.is_type("keyword") and seg.name == "SELECT" and select_idx - 1:
+            if seg.is_type("keyword") and seg.name == "SELECT" and select_idx == -1:
```
Comment (barrywhart) [inline src/sqlfluff/core/rules/std.py:3465]: This is not directly related to the original issue, but the code just looks wrong. I'm pretty sure the intent was to check for a `-1` value, i.e. "not yet set" just as with `first_new_line_idx`.

File: src/sqlfluff/core/rules/std.py [hunk 2]
```diff
@@ -3462,11 +3462,14 @@ def _get_indexes(segment):
...
                 select_idx = fname_idx
             if seg.is_type("newline") and first_new_line_idx == -1:
                 first_new_line_idx = fname_idx
-            if seg.is_type("whitespace") and first_whitespace_idx == -1:
+            # TRICKY: Ignore whitespace prior to the first newline, e.g. if
+            # the line with "SELECT" (before any selct targets) has trailing
+            # whitespace.
+            if seg.is_type("whitespace") and first_new_line_idx != -1 and first_whitespace_idx == -1:
```
Comment (barrywhart) [inline src/sqlfluff/core/rules/std.py:3472]: This is the real fix. By ignoring whitespace prior to the first newline, we avoid being fooled by trailing whitespace after the `SELECT`.

Question: Does this potentially cause other issues?

File: test/core/rules/test_cases/L036.yml [hunk 3]
```diff
@@ -22,7 +22,11 @@ test_single_select_target_and_new_line_between_select_and_select_target:
         *
     from x
 
-test_mulitple_select_targets_all_on_the_same_line:
+test_multiple_select_targets_all_on_the_same_line:
```
Comment (barrywhart) [inline test/core/rules/test_cases/L036.yml:25]: Fixing a typo: mulitple -> multiple

**Comment by dmateusp** [comment]
I'm wondering if during the parse we couldn't tag the trailing whitespaces as such (maybe with an attribute `is_trailing` ?), to make it easier to ignore them when fixing the rules.
I imagine dealing with trailing whitespaces could be an issue in other rules

**Comment by barrywhart** [comment]
@dmateusp: I agree -- either do that during the parse or provide a reusable function somewhere, depending on how common and consistent the logic is.

**Review by NiallRees — APPROVED** [review]
